docs/build/build_system.md

```diff
@@ -0,0 +1,213 @@
+# Build system
+
+## Implementation details
+
+The project uses CMake as the build system generator. CMake allows for cross-platform builds and provides the necessary flexibility for both project development and its use in external projects.
+
+[Conan](https://conan.io) is integrated as an optional dependency manager via dependency resolver mechanism introduced in CMake 3.24 (see `cmake/conan_provider.cmake`). Conan helps us manage external libraries and allows us to create reproducible developer environment with ease. However, the build system is designed to work without Conan if needed.
+
+## Compilation steps
+
+Typical configurations of the project are provided in the `CMakePresets.json` file – this means that if you use a modern IDE, **no additional configuration is needed**. Just pick one of the configurations provided by this file and use controls in your IDE to recompile and run tests (should work with VSCode + CMake Tools extension, JetBrains CLion and probably many others).
+
+If you wish to use CLI instead, you can list all the available configurations in the `CMakePresets.json` file and pick one of the presets:
+
+```sh
+cmake --list-presets
+```
+
+!!! warning
+    By default, the standard repositories for Ubuntu 22.04 provide CMake version 3.22, which does not support certain features required for working with CMakePresets.json. To avoid issues during the build process, you need to manually install CMake version 3.24 or higher.
```

Comment:
Since people read sequentially it makes sense to move the warning before the first cmake invocation. I've reworded the warning a bit so that it'd be more helpful to beginners.
```suggestion
## Compilation steps

!!! warning
    Our build system settings require using CMake 3.24 or newer. Make sure your version is compatible by running `cmake --version` and use the [CMake installation guide](https://cliutils.gitlab.io/modern-cmake/chapters/intro/installing.html) to upgrade to latest version if yours is older than 3.24. The universal method is to use `pipx install cmake` (don't forget to run `pipx ensurepath` and relaunching your shell after installation), but you may choose reconfiguring your OS's package manager instead.

Typical configurations of the project are provided in the `CMakePresets.json` file – this means that if you use a modern IDE, **no additional configuration is needed**. Just pick one of the configurations provided by this file and use controls in your IDE to recompile and run tests (should work with VSCode + CMake Tools extension, JetBrains CLion and probably many others).

If you wish to use CLI instead, you can list all the available configurations in the `CMakePresets.json` file and pick one of the presets:

```sh
cmake --list-presets
```
```